pkgs/stdenv/generic/setup.sh

```diff
@@ -530,15 +538,14 @@ _activatePkgs
 # with this information to the relevant env hook array, but bash
 # doesn't have closures, so it's easier to just pass this in.
 _addToEnv() {
-    local -i depHostOffset depTargetOffset
-    local pkg
+    local depHostOffset depTargetOffset pkg
 
     for depHostOffset in "${allPlatOffsets[@]}"; do
-        local hookVar="${pkgHookVarVars[$depHostOffset + 1]}"
-        local pkgsVar="${pkgAccumVarVars[$depHostOffset + 1]}"
+        local hookVar="${pkgHookVarVars[$(( depHostOffset + 1 ))]}"
```

Comment:
ditto here and throughout, $(( )) is optional in a bash/ksh array subscript

Reply:
Thank you for this one, I'll keep this in mind!